### PR TITLE
Add few warning compiler options to error

### PR DESCRIPTION
**Pull request by saikishor**
Inspiring from the PRs that @destogl made recently: https://github.com/ros-controls/ros2_control/pull/1173 & https://github.com/ros-controls/ros2_control/pull/1174. I think it is in the best interest of everyone to add some compiler flags to errors to better code quality. I have done it for the most important packages and fixed some errors at the same time

### CODE REVIEW

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -1113,21 +1109,21 @@ controller_interface::return_type ControllerManager::switch_controller(
...
       in_activate_list = false;
       activate_request_.erase(activate_list_it);
     }
 
     const auto extract_interfaces_for_controller =
-      [this](const ControllerSpec controller, std::vector<std::string> & request_interface_list)
+      [this](const ControllerSpec ctrl, std::vector<std::string> & request_interface_list)
     {
-      auto command_interface_config = controller.c->command_interface_configuration();
+      auto command_interface_config = ctrl.c->command_interface_configuration();
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp:1126]: We seldom shorten "controller" and even if we did it should be consistent. I don't think you gain much with this shortening here

Comment (saikishor) [reply controller_manager/src/controller_manager.cpp:1126]: The problem here is the `controller` variable was shadowing a different `controller` variable above. That's the reason, I had to change the name of the variables to a different one if not it doesn't compile after adding shadow option

Comment (bmagyar) [reply controller_manager/src/controller_manager.cpp:1126]: Right! Thanks for the explanation

**Review by bmagyar — APPROVED** [review]
Thank you sir!

**Comment by christophfroehlich** [comment]
@mergifyio backport humble iron

**Pull request merged**